mycore-iview2/src/main/java/org/mycore/iview2/frontend/MCRIView2XSLFunctions.java

```
@@ -63,20 +63,17 @@ public static String getOptions(String derivateID, String extensions) {
      *         null if there are no derivates related to the given mcrid
      */
     public static String getSupportedMainFileByOwner(String mcrID) {
-        MCRObjectID objectID = null;
-
         try {
-            objectID = MCRObjectID.getInstance(mcrID);
+            MCRObjectID objectID = MCRObjectID.getInstance(mcrID);
+            MCRObject obj = MCRMetadataManager.retrieveMCRObject(objectID);
+            List<MCRMetaEnrichedLinkID> derivates = obj.getStructure().getDerivates();
+            if (!derivates.isEmpty()) {
+                return derivates.getFirst() + "/" + ADAPTER.getSupportedMainFile(derivates.getFirst().toString());
+            }
+            return null;
         } catch (Exception e) {
             return null;
         }
```

Looks strange, but as it is intended for xsl, it should be ok.

---

Looks indeed "strange", but with PMD we cant' have empty catch blocks .